word2vec.py

```diff
@@ -0,0 +1,73 @@
+import re
+import string
+
+import gensim
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from pycocotools.coco import COCO
+
+
+def sentence2vec(sentence, model):
+    ''' Converts a sentence to a vector by averaging each word vector
+    representation given by the model '''
+    # Remove all non alphanumeric characters
+    regex = re.compile('[\W_ ]+') # TODO: compiling once for all calls might be faster
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
+    sentence = regex.sub('', sentence)
+    words = sentence.split(' ')
+
+    # Average vector representations of the words in the sentence
+    vec = np.zeros(model.vector_size)
+    count = 0
+    for word in words:
+        if word in model.vocab:
+            vec += model[word]
+            count += 1
+    vec /= count
+    return vec
+
+def create_caption_dataframe():
+    ''' Create a dataframe containing all captions for each images '''
+    # initialize COCO api for caption annotations
+    dataDir='dataset'
+    dataType='train2014'
+    annFile = '%s/annotations/captions_%s.json'%(dataDir,dataType)
+    coco = COCO(annFile)
+
+    # Get all image ids
+    img_ids = coco.getImgIds()
+
+    # An image can have several captions
+    # Concatenate the captions for each image and save as csv
+    df = pd.DataFrame(index=sorted(img_ids), columns=['caption'])
+    for img_id in tqdm(img_ids, desc='Retrieving captions'):
+        ann_ids = coco.getAnnIds(imgIds=img_id)
+        anns = coco.loadAnns(ann_ids)
+        caption = ''
+        for ann in anns:
+            caption += ' ' + ann['caption']
+        df.loc[img_id] = caption
+    df.to_csv('%s/annotations/captions.csv'% dataDir)
```

Comment:
Ouais bien sûr ! Comme je te dis pour moi dans la code review tu notes tout ce qui te semble bizarre ou à optimiser. C'est pas du tout un reproche, c'est juste comme ça on est sur la même longueur d'ondes à chaque itération !